### PR TITLE
Use OpenFile for local files

### PR DESCRIPTION
**Pull request by matthiasclasen**
The OpenURI portal method no longer accepts file: uris.

### CODE REVIEW

File: src/xdg-open.c [hunk 1]
```diff
@@ -92,19 +97,53 @@ main (int argc, char *argv[])
...
-                               &error);
+  file = g_file_new_for_commandline_arg (uris[0]);
+  if (g_file_is_native (file))
+    {
+      char *path;
+      int fd;
+      GUnixFDList *fd_list;
+
+      path = g_file_get_path (file);
+      fd = open (path, O_PATH | O_CLOEXEC);
```
Comment (alexlarsson) [inline src/xdg-open.c]: You need some error checking here.

File: src/xdg-open.c [hunk 2]
```diff
@@ -45,6 +48,8 @@ main (int argc, char *argv[])
   GError *error = NULL;
   GDBusConnection *connection;
   GVariantBuilder opt_builder;
+  GFile *file;
+  char *uri;
```
Comment (fmuellner) [inline src/xdg-open.c]: Unused variable - either remove it, or use

  const char *uri = uris[0];

to keep the code that ignores any uri but the first one in a single place ...

**Comment by alexlarsson** [comment]
Looks good to me.

**Comment by fmuellner** [comment]
Looks good to me as well, thanks!

**Comment by alexlarsson** [comment]
I added this to the freedesktop 1.6 sdk

**Pull request merged**